### ISSUE TITLE
Cannot use member function pointer to non-overridden inherited function when registering methods

### ISSUE THREAD

**Issue by raymoo** [godotengine/godot-cpp#303]
It seems like when registering a method for a class, if you give the call a member function pointer from a base class, it will not register the method for the current class. I think it registers it for the base class.

It doesn't work even if you use the current class's namespace to get the pointer. For example, if `D` derives from `B`, and `B` defines `_process` (but `D` does not), then `register_method("_process", &D::_process)` will register the method for `B`, not for `D`.

Here is how I'm actually using it:
* [godot/object.h](https://paste.debian.net/1087304/)
* [godot/enemy.h](https://paste.debian.net/1087305/)
* [godot/enemy.cpp](https://paste.debian.net/1087306/)

I believe that this issue may have the same cause as https://github.com/GodotNativeTools/godot-cpp/issues/256

**Comment by raymoo** [comment]
Just to note, my code works if I define a new `_process` in my derived class and use it to call the base's `_process`. However I want to avoid defining a new `_process` in the derived class because I want the base class to control `_process`.

**Comment by raymoo** [comment]
Here's a definition to use as a workaround / potentially to add to godot-cpp itself:
```
template <class D, class B, class R, class... As>
void register_class_method(const char *name, R (B::* method_ptr)(As...), godot_method_rpc_mode rpc_type = GODOT_METHOD_RPC_MODE_DISABLED)
{
    R (D::* casted_ptr)(As...) = method_ptr;
    register_method(name, casted_ptr, rpc_type);
}
```

Usage:
```
register_class_method<DerivedClass>("_process", &BaseClass::_process);
```

**Comment by Zylann** [comment]
Addressed by #304
